plugin.py

```diff
@@ -33,38 +34,45 @@ def install_in_cache(cls) -> bool:
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (12, 0, 0)
 
-    @classmethod
-    def on_settings_read(cls, settings: sublime.Settings) -> bool:
-        super().on_settings_read(settings)
+    def on_settings_changed(self, settings: DottedDict) -> None:
+        super().on_settings_changed(settings)
 
-        if settings.get("dev_environment") == "sublime_text":
-            server_settings = settings.get("settings", {})  # type: Dict[str, Any]
+        if self.get_plugin_setting("dev_environment") == "sublime_text":
```

Comment:
This might sound like a crazy idea but this `get_plugin_setting` wouldn't be necessary had we put our `dev_environment` inside the `settings` object.

The server will ignore unknown properties anyway and we'll get the benefit of merged settings automatically (not that important in this case but could be in other cases).

Noticed that it's already happening as the `python` settings are not really pyright specific in VSCode and the server still receives all of them.

Reply:
Not against that but probably adding an extra prefix like `xxx.dev_environment`?

Reply:
I don't feel it's really necessary. We should treat those settings as ours as that's how normal users see them. Users don't need to know the distinction between our and original pyright settings.

Reply:
having a dedicated prefix makes it more structured imho. now we only have one settings though.

Reply:
I'm fine with a prefix but not sure what it would be.

Reply:
I propose using the package name as the prefix, i.e., `LSP-pyright.dev_environment`.

Reply:
Sorry but it sounds very redundant to put a `LSP-pyright.*` key inside the `LSP-pyright.sublime-settings` or inside the `LSP-pyright` object in project settings, for example.

I think it only achieves peace of mind for you, and possibly other developers working on the plugin, because you want to keep the settings of pyright and our additions separate. But as I said, I don't believe those should be treated as separate. To normal users, there is no distinction really.

That setting is a bit special as it affects/modifies other settings though. So from that perspective, I guess it might make sense to single it out somehow. But I really can't think of anything better right now. Personally, I would think that a name like `environment_type` would be enough.

Reply:
> That setting is a bit special as it affects/modifies other settings though. So from that perspective, I guess it might make sense to single it out somehow.

Does it imply only `dev_environment` (whatever its name) will be in server settings but not other non-server settings in the future?

Reply:
I don't think that implies anything as we don't have strict rules for this. I'm just trying to think of what's best to do here.

But in the end, it's a trivial thing and we should probably not spend too much time discussing it.

Maybe just keep as it is now, even if it complicates the code internally...

Reply:
Yeah. I guess that can be discussed later if wanted. https://github.com/neoclide/coc-python#options seems to just use `python` as the prefix.